extensions/resteasy/runtime/src/main/java/io/quarkus/resteasy/runtime/AuthenticationFailedExceptionMapper.java

```diff
@@ -0,0 +1,46 @@
+package io.quarkus.resteasy.runtime;
+
+import javax.annotation.Priority;
+import javax.enterprise.inject.spi.CDI;
+import javax.ws.rs.Priorities;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.ext.ExceptionMapper;
+import javax.ws.rs.ext.Provider;
+
+import io.quarkus.security.AuthenticationFailedException;
+import io.quarkus.vertx.http.runtime.CurrentVertxRequest;
+import io.quarkus.vertx.http.runtime.security.ChallengeData;
+import io.quarkus.vertx.http.runtime.security.HttpAuthenticator;
+import io.vertx.ext.web.RoutingContext;
+
+@Provider
+@Priority(Priorities.USER + 1)
+public class AuthenticationFailedExceptionMapper implements ExceptionMapper<AuthenticationFailedException> {
```

Comment:
Why is this mapper needed ?

Reply:
I think `ForbiddenException` and/or `UnauthorizedException` may be thrown on some JAX-RS paths too

Reply:
This is needed because if you use the @RolesAllowed interceptor with a deferred identity you may get this exception when resolving the deferred identity. This mapper will handle it correctly.

By default the deferred identity will automatically handle this on a vert.x level, however we have to disable this for JAX-RS as you get an error if the vert.x response is closed while the JAX-RS request is ongoing.

Reply:
@stuartwdouglas OK, makes sense